File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int cp = prices[0];

        int profit = 0;
        int maxProfit = 0;

        for(int i=1; i<n; i++) {
            if(prices[i] < cp) {
                cp = prices[i];
            }

            profit = prices[i] - cp;
            maxProfit = Math.max(profit, maxProfit);
        }

        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int buyPrice = prices[0];
        int ans = 0;

        for(int i=1; i<n; i++) {
            if(prices[i] < buyPrice) {
                buyPrice = prices[i];
            } else {
                ans = max(ans, prices[i]-buyPrice);
            }
        }
        return ans;
    }
};